src/Symfony/Component/Runtime/SymfonyRuntime.php

```diff
@@ -0,0 +1,194 @@
+ *
+ * Accepts the following options:
+ *  - "debug" to toggle debugging features
+ *  - "env" to define the name of the environment the app runs in
+ *  - "disable_dotenv" to disable looking for .env files
+ *  - "dotenv_path" to define the path of dot-env files - defaults to ".env"
+ *  - "prod_envs" to define the names of the production envs - defaults to ["prod"]
+ *  - "test_envs" to define the names of the test envs - defaults to ["test"]
+ *
+ * When these options are not defined, they will fallback:
+ *  - to reading the "APP_DEBUG" and "APP_ENV" environment variables;
+ *  - to parsing the "--env|-e" and "--no-debug" command line arguments
+ *    if the "symfony/console" component is installed.
+ *
+ * When the "symfony/dotenv" component is installed, .env files are loaded.
+ * When "symfony/error-handler" is installed, it is registred in debug mode.
+ *
+ * On top of the base arguments provided by GenericRuntime,
+ * this runtime can feed the app-callable with arguments of type:
+ *  - Request from "symfony/http-foundation" if the component is installed;
+ *  - Application, Command, InputInterface and/or OutputInterface
+ *    from "symfony/console" if the component is installed.
+ *
+ * This runtime can handle app-callables that return instances of either:
+ *  - HttpKernelInterface,
+ *  - Response,
+ *  - Application,
+ *  - Command,
+ *  - int|string|null as handled by GenericRuntime.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @experimental in 5.3
+ */
+class SymfonyRuntime extends GenericRuntime
+{
+    private $input;
+    private $output;
+    private $console;
+    private $command;
+    private $env;
+
+    public function __construct(array $options = [])
+    {
+        $this->env = $options['env'] ?? null;
+        $_SERVER['APP_ENV'] = $options['env'] ?? $_SERVER['APP_ENV'] ?? null;
+
+        parent::__construct($options);
+
+        if (isset($_SERVER['argv']) && null === $this->env && class_exists(ArgvInput::class)) {
+            $this->getInput();
+        }
+
+        if (($options['disable_dotenv'] ?? false) || !isset($options['project_dir']) || class_exists(MissingDotenv::class, false)) {
+            $_SERVER['APP_DEBUG'] = filter_var($options['debug'] ?? $_SERVER['APP_DEBUG'] ?? true, \FILTER_VALIDATE_BOOLEAN) ? '1' : '0';
+        } else {
+            (new Dotenv())
+                ->setProdEnvs((array) ($options['prod_envs'] ?? ['prod']))
+                ->bootEnv($options['project_dir'].'/'.($options['dotenv_path'] ?? '.env'), 'dev', (array) ($options['test_envs'] ?? ['test']));
+        }
+
+        if ($_SERVER['APP_DEBUG'] && class_exists(Debug::class)) {
+            umask(0000);
+            Debug::enable();
+        }
+    }
+
+    public function getRunner(object $application): RunnerInterface
+    {
+        if ($application instanceof HttpKernelInterface) {
+            return new HttpKernelRunner($application, Request::createFromGlobals(), $this);
```

Comment:
The third parameter is never used. 
```suggestion
            return new HttpKernelRunner($application, Request::createFromGlobals());
```

Reply:
good catch thanks